### PR TITLE
Fix invalid session creation when app is launched in background

### PR DESCRIPTION
**Pull request by markushi**
## :scroll: Description
Skip session creation if the app is not started in foreground.

## :bulb: Motivation and Context
This is a follow-up based on some customer feedback, specifically seeing session creations spikes for scheduled background jobs (e.g. a nightly sync).

Likely fixes https://github.com/getsentry/sentry-java/discussions/2512 

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by markushi** [comment]
@romtsn as an alternative we could keep the old logic (always create a session) and attach a foreground flag once the first Activity is launched ~ `session.setIsForeground(true|false)`. When `SentryClient.captureSession()` is called, we could then drop non-foreground sessions. This way we don't have to check for foreground importance which might causes issues in third-party app stores (see https://github.com/getsentry/sentry-java/pull/2537)

**Comment by romtsn** [comment]
> @romtsn as an alternative we could keep the old logic (always create a session) and attach a foreground flag once the first Activity is launched ~ `session.setIsForeground(true|false)`. When `SentryClient.captureSession()` is called, we could then drop non-foreground sessions. This way we don't have to check for foreground importance which might causes issues in third-party app stores (see #2537)

I think that makes sense, would you mind changing it then?

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -479,6 +479,13 @@ public void captureSession(final @NotNull Session session, final @Nullable Hint
       return;
     }
 
+    if (session.isAppInForeground() != null && !session.isAppInForeground()) {
```
Comment (stefanosiano) [inline sentry/src/main/java/io/sentry/SentryClient.java:482]: Maybe replace this with
`if (!Boolean.TRUE.equals(session.isAppInForeground()))`?

**Review by romtsn — APPROVED** [review]

**Comment by romtsn** [comment]
:shipit:

**Pull request merged**